src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessCollectionTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Diagnostics.ProcessTests
+{
+    public class ProcessCollectionTests : ProcessTestBase
+    {
+        [Fact]
+        public void TestModuleCollectionBehavior()
+        {
+            ProcessModule[] mArray = _process.Modules.Cast<ProcessModule>().ToArray();
+
+            ProcessModuleCollection moduleCollection = new ProcessModuleCollection(mArray);
+            Assert.True(moduleCollection.Count > 1);
+            Assert.Equal(mArray[0], moduleCollection[0]);
```

Comment:
Shouldn't we verify that every element in the module collection matches every corresponding element in mArray, not just the first?